contracts/src/v0.8/automation/dev/v2_3/AutomationRegistrar2_3.sol

```diff
@@ -362,26 +308,12 @@ contract AutomationRegistrar2_3 is TypeAndVersionInterface, ConfirmedOwner, IERC
    * @param amount Amount of LINK sent (specified in Juels)
    * @param data Payload of the transaction
    */
-  function onTokenTransfer(
-    address sender,
-    uint256 amount,
-    bytes calldata data
-  )
-    external
-    override
-    onlyLINK
-    permittedFunctionsForLINK(data)
-    isActualAmount(amount, data)
-    isActualSender(sender, data)
-  {
-    if (amount < s_config.minLINKJuels) {
-      revert InsufficientPayment();
-    }
-    (bool success, ) = address(this).delegatecall(data);
-    // calls register
-    if (!success) {
-      revert RegistrationRequestFailed();
-    }
+  function onTokenTransfer(address sender, uint256 amount, bytes calldata data) external override {
+    if (msg.sender != address(LINK)) revert OnlyLink();
+    RegistrationParams memory params = abi.decode(data, (RegistrationParams));
+    if (address(params.billingToken) != address(LINK)) revert OnlyLink();
```

Comment:
when is this onTokenTransfer called? This takes only LINK?

Reply:
yeah this is an ERC677 function - it can only be called by the link token